homeassistant/components/lcn/__init__.py

```diff
@@ -120,26 +138,69 @@ async def async_setup(hass, config):
             DOMAIN, service_name, service(hass).async_call_service, service.schema
         )
 
+    if DOMAIN not in config:
+        return True
+
+    # initialize a config_flow for all LCN configurations read from
+    # configuration.yaml
+    config_entries_data = import_lcn_config(config[DOMAIN])
+
+    for config_entry_data in config_entries_data:
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN,
+                context={"source": config_entries.SOURCE_IMPORT},
+                data=config_entry_data,
+            )
+        )
     return True
 
 
+async def async_entry_updated(hass, config_entry):
+    """Update listener to change connection name."""
+    hass.data[DOMAIN][config_entry.entry_id][
+        CONNECTION
+    ].connection_id = config_entry.title
+
+
 class LcnEntity(Entity):
-    """Parent class for all devices associated with the LCN component."""
+    """Parent class for all entities associated with the LCN component."""
 
-    def __init__(self, config, device_connection):
+    def __init__(self, config, host_id, device_connection):
         """Initialize the LCN device."""
         self.config = config
+        self.host_id = host_id
         self.device_connection = device_connection
+        self._unregister_for_inputs = None
         self._name = config[CONF_NAME]
 
+    @property
+    def unique_id(self):
+        """Return a unique ID."""
+        unique_device_id = generate_unique_id(
+            (
+                self.device_connection.seg_id,
+                self.device_connection.addr_id,
+                self.device_connection.is_group,
+            )
+        )
+        return f"{self.host_id}-{unique_device_id}-{self.config[CONF_RESOURCE]}"
```

Comment:
What is the `host_id` for kind of value? If it's the config entry entry_id we should keep calling it that.